pydsx/singly_linked_list/linked_list.py

```diff
@@ -172,6 +175,148 @@ def __setitem__(self, index, new_value):
                 raise IndexError("Index out of range")
             current_node = current_node.next
         current_node.value = new_value
+    
+    def sort(self, reverse = False, key = "merge"):
+        """
+        Sorts the SinglyLinkedList object it's called on.
+
+        Args:
+            reverse: Flag to reverse the sorting order, is set to false by
+            default.
+            key: parameter for choosing the sorting algorithm.
+
+        Returns:
+            Sorted SinglyLinkedList object.
+        
+        """
+        if reverse:
+            if self._head == None:
+                raise IndexError("The linked list is empty")       
+            if self._head.next == None:
+                return self 
+            if key == "merge":
+                return self.merge_sort(reverse)
+            elif key == "quick":
+                pass
+            elif key == "heap":
+                pass
+            else:
+                raise ValueError("Unknown sorting method specified")
+        else:
+            if self._head == None:
+                raise IndexError("The linked list is empty")       
+            if self._head.next == None:
+                return self 
+            if key == "merge":
+                return self.merge_sort()
+            elif key == "quick":
+                pass
+            elif key == "heap":
+                pass
+            else:
+                raise ValueError("Unknown sorting method specified")
```

Comment:
I was looking at maybe we can try to apply a (DRY, don't reapt yourself reference for DRY (refactoring book))

May you can extract to a  method?
something like this

```python
self._sort_strategy(reverse, key)
#and self._merge_sort(...) choose if reversed or not
def _sort_strategy(self, reverse, key):
    if self._head == None:
        raise IndexError("The linked list is empty")       
    if self._head.next == None:
        return self 
    if key == "merge":
        return self.merge_sort(reverse)
    elif key == "quick":
        pass
    elif key == "heap":
        pass
    else:
       raise ValueError("Unknown sorting method specified")
       
```
Does that make sense to you?

I'll keep reviewing, thanks !!
Congratulations on the code

Reply:
That does make sense, I changed it up while trying to fix a bug in the  ```merge_sort``` method, I'll put it back to how it was.